fabtests/component/sock_test.c

```diff
@@ -48,338 +48,380 @@
 
 #include <shared.h>
 
-int *clients;
 
-fd_set select_set;
-int max_sock, connections;
-struct pollfd *poll_set;
-struct epoll_event *events;
-int efpd;
+static int *fds;
+static int connections = 1000;
+uint64_t starttime, endtime;
 
-static int start_server()
+static struct pollfd *poll_set;
+static struct epoll_event *ep_events;
+static int epfd = -1;
+
+
+static int start_server(void)
 {
 	struct addrinfo *ai = NULL;
-	int ret;
 	int optval = 1;
+	int ret;
 
 	ret = getaddrinfo(opts.src_addr, opts.src_port, NULL, &ai);
 	if (ret) {
-		printf("getaddrinfo failed\n");
-		goto out;
+		FT_PRINTERR("getaddrinfo", ret);
+		return -ret;
 	}
 
 	listen_sock = socket(ai->ai_family, SOCK_STREAM, 0);
 	if (listen_sock < 0) {
-		printf("socket error: %i: %s\n", errno, strerror(errno));
-		ret = -1;
-		goto out;
+		FT_PRINTERR("socket", -errno);
+		goto free;
 	}
 
-	ret = setsockopt(listen_sock, SOL_SOCKET, SO_REUSEADDR, (char *) &optval, 
-		sizeof(optval));
+	ret = setsockopt(listen_sock, SOL_SOCKET, SO_REUSEADDR,
+			 (char *) &optval, sizeof(optval));
 	if (ret) {
-		printf("error setting socket options\n");
-		goto out;
+		FT_PRINTERR("setsockopt", -errno);
```

Comment:
Is there a risk that FT_PRINTESS overwrites `errno`? Because it's used as return code line 105.

Reply:
Yes it could.  It would be best to update the macro to save/restore errno, similar to what we do with FI_LOG in libfabric.  I'll create a separate patch for that.